traitsui/qt4/ui_panel.py

```diff
@@ -204,11 +204,35 @@ def __init__(self, ui, parent, is_subpanel):
                         self.add_button(button, bbox, role)
                 layout.addWidget(bbox)
 
+        # If the UI has a toolbar, should add it to the panel too
+        if self.ui.view.toolbar:
+            self._add_toolbar(parent)
+
         # Ensure the control has a size hint reflecting the View specification.
         # Yes, this is a hack, but it's too late to repair this convoluted
         # control building process, so we do what we have to...
         self.control.sizeHint = _size_hint_wrapper(self.control.sizeHint, ui)
 
+    def _add_toolbar (self, parent):
+        """ Adds a toolbar to the `parent` (QtWindow)
+        """
+        toolbar = self.ui.view.toolbar
+        if toolbar is not None:
+            self._last_group = self._last_parent = None
+            qt_toolbar = toolbar.create_tool_bar(parent, self )
+            qt_toolbar.setMovable( False )
+            parent.addToolBar( qt_toolbar )
+            self._last_group = self._last_parent = None
+
+    def can_add_to_toolbar(self, action):
```

Comment:
Is this function used anywhere?  It looks like it is needed to hide items which shouldn't be displayed based on the View's context, but I don't see it being explicitly called.